Week-1/how-web-works/how-web-works.md

```diff
@@ -0,0 +1,116 @@
+## What is the main functionality of a browser ? ##
+
+A browser uses a URI (Uniform Resource Identifier) and sends requests for data(HTML/CSS/JS/XML/JSON and others) to other nodes(server/another computer) in its network and serve received data to the requested user.
+
+A browser uses multiple protocols (Set of rules) & ports (80 : unsecure & 443 : secure) to send & receive different types of resources in its network and display them in a presentable way to the end-user.
+
+## High Level Components of a browser ##
+
+The important high level components of a browser are as follows:
+
+![components of browser](./images/components.jpg)
+
+1. User Interface:
+
+User interface contains the following sub components.
+address bar, forward/back buttons, bookmarks, window to show content etc. 
+
+2. Browser Engine:
+
+This works with both rendering engine and user interface to show the final output.
+To display non-native files like PDF,DOC etc... browser needs plugins/extensions.
+
+3. Rendering Engine:
+
+This is responsible for parsing HTML, CSS, scripts etc., and displaying them on screen.
+
+4. Networking:
+
+This sends network requests and handles incoming data from other computers (servers) in the network.
+The URI is translated into an IP address (exact location of the server that hosts the requested resources) by DNS server. Then the request is sent to that server. 
+Port 80 is used by default for unsecure communications.
+Port 443 is used for secure connections. 
+
+5. UI Backend:
+
+Used to draw basic widgets like combo boxes and windows that are platform independent.
+
+6. Javascript Interpreter:
+
+It parses JS code and executes it.
+
+7. Data Storage:
+
+This is where data is stored locally for persistence (Eg. cookies)
+Browsers support storage mechanisms such as localStorage, IndexedDB, WebSQL and FileSystem
+
+## Rendering Engine and its Use ##
+
+![webkit rendering engine](./images/webkitflow-renderingengine.png)
+
+Rendering engine interacts with network layer to get documents(HTML/CSS etc) in chunks of 8kbs.
```

Comment:
Can you provide documentation that supports this ?

Reply:
Yes.  https://www.html5rocks.com/en/tutorials/internals/howbrowserswork/#The_main_flow
Added this link at the bottom.